### PR TITLE
Initial Chart

### PR DESCRIPTION
**Pull request by malacourse**
This works with the assemble backstage image.

### CODE REVIEW

File: charts/backstage/values.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+# name of instances
+name: backstage
+
+postgres:
+  external: false
+  ## Use presistent storage
+  storage: 
+    enabled: true
+    size: 2G
```
Comment (sabre1041) [inline charts/backstage/values.yaml:34]:
```suggestion
    size: 2Gi
```

File: charts/backstage/values.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
+GreplicaCount: 1
```
Comment (tumido) [inline charts/backstage/values.yaml:1]: Typo, variable name used in `Deployment` is `replicaCount`.

```suggestion
replicaCount: 1
```

File: charts/backstage/values.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  className: ""
+
+  ## @param ingress.annotations Additional annotations for the Ingress resource
+  annotations: 
+    nginx.ingress.kubernetes.io/rewrite-target: "/"
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+    route.openshift.io/termination: "edge"
+
+  ## @param host Hostname to be used to expose the route to access the backstage application (e.g: backstage.IP.nip.io)
+  host: ""
```
Comment (tumido) [inline charts/backstage/values.yaml:36]: I think `host` value should be the same/match `.backstage.baseUrl` shouldn't it? If they differ backstage won't be able to perform auth properly/cors properly.

https://backstage.io/docs/auth/troubleshooting#auth-fails-with-login-failed-caused-by-notallowederror-origin--is-not-allowed

File: charts/backstage/templates/app-config.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        # on your specific deployment. If you remove the host line entirely, the
+        # backend will bind on the interface that corresponds to the backend.baseUrl
+        # hostname.
+        host: 0.0.0.0
+      csp:
+        connect-src: ["'self'", 'http:', 'https:']
+        # Content-Security-Policy directives follow the Helmet format: https://helmetjs.github.io/#reference
+        # Default Helmet Content-Security-Policy values can be removed by setting the key to false
+      cors:
+        origin: http://{{ .Release.Name }}:3000
```
Comment (tumido) [inline charts/backstage/templates/app-config.yaml:36]:
```suggestion
        origin: {{ required "Provide App baseUrl in values" .Values.backstage.baseUrl }}
```

**Review by tumido — CHANGES_REQUESTED** [review]

File: charts/backstage/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  name: {{ include "backstage.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: 
+    {{- include "backstage.labels" . | nindent 4 }}
+{{- if .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml .Values.ingress.annotations | nindent 4 }}
+{{- end }}
+spec:
```
Comment (raffaelespazzoli) [inline charts/backstage/templates/ingress.yaml:13]: do we want TLS here?

File: charts/backstage/templates/route.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.route.enabled }}
+apiVersion: route.openshift.io/v1
+kind: Route
```
Comment (raffaelespazzoli) [inline charts/backstage/templates/route.yaml:3]: we already have ingress...do we also need route? I think it's either one or the other. This route looks better as it had TLS.

File: charts/backstage/values.yaml [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+ingress:
+  enabled: false
+
+  ## @param ingress.className Name of the IngressClass cluster resource which defines which controller will implement the resource (e.g nginx)
+  className: ""
+
+  ## @param ingress.annotations Additional annotations for the Ingress resource
+  annotations: 
+    nginx.ingress.kubernetes.io/rewrite-target: "/"
```
Comment (raffaelespazzoli) [inline charts/backstage/values.yaml:31]: nginx?

File: charts/backstage/templates/route.yaml (hunk 6 above)
Comment (malacourse) [reply charts/backstage/templates/route.yaml:3]: Leaving both, ingress is TLS due to the annotation: route.openshift.io/termination: "edge"

File: charts/backstage/templates/ingress.yaml (hunk 5 above)
Comment (malacourse) [reply charts/backstage/templates/ingress.yaml:13]: route.openshift.io/termination: "edge"
 create TLS

File: charts/backstage/values.yaml (hunk 3 above)
Comment (malacourse) [reply charts/backstage/values.yaml]: Modified to only use the baseUrl for all host vars

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**